code/modules/unit_tests/json_logging.dm

```diff
@@ -0,0 +1,17 @@
+/// Currently this only checks for a valid semver string
+/datum/unit_test/json_logging
+
+/datum/unit_test/json_logging/Run()
+	for(var/datum/log_entry/entry as anything in subtypesof(/datum/log_entry))
+		var/log_version = initial(entry.version)
+		// check for invalid log version
+		if(!log_version)
+			TEST_FAIL("[entry] has no log version; how did you mess this up?")
+			continue
+		var/regex/version_regex = regex("^[0-9]+\\.[0-9]+\\.[0-9]+$")
+		if(!version_regex.Find(log_version))
+			TEST_FAIL("[entry] has an invalid log version; it must be a semver version string")
+			continue
+	var/base_log_version = /datum/log_entry:log_version
+	if(base_log_version != "0.0.0")
```

Comment:
Wait wat? This isn't what I wanted the test to be, I want to just make sure OTHER people don't keep the default. "0.0.0" cannot be acceptable as a log version.

Reply:
This is checking the base value, to make sure people aren't modifying the base datum

Reply:
I guess but can't we do that just with a comment, or don't fill it in at all